extensions/wikia/Helios/HelperController.class.php

```diff
@@ -127,4 +131,57 @@ public function sendConfirmationEmail()
 
 		$this->response->setVal( 'success', true );
 	}
+
+	public function isBlocked() {
+		if ( !$this->authenticateViaTheSchwartz() ) {
+			return;
+		}
+
+		$username = $this->getFieldFromRequest( 'username', 'invalid username' );
+		if ( !isset( $username ) ) {
+			return;
+		}
+
+		$blocked = $this->getAuthService()->isUsernameBlocked( $username );
+		if ( $blocked === null ) {
+			$this->response->setVal( 'message', 'user not found' );
+			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_NOT_FOUND );
+			return;
+		}
+
+		$this->response->setData( array( 'blocked' => $blocked ) );
+	}
+
+	protected function getFieldFromRequest( $field, $failureMessage ) {
+		$username = $this->getVal( 'username', null );
+		if ( !isset( $username ) ) {
+			$this->response->setVal( 'message', $failureMessage );
+			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+		}
+
+		return $username;
+	}
+
+	protected function authenticateViaTheSchwartz() {
+		if ( $this->getVal( 'secret' ) != $this->wg->TheSchwartzSecretToken ) {
+			$this->response->setVal( 'message', 'invalid secret' );
+			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_FORBIDDEN );
+			return false;
+		}
+
+		return true;
+	}
+
+	public function setAuthService( \Wikia\Service\User\Auth\AuthService $authService ) {
```

Comment:
Why do we use the fully qualified class name here but not below? seems like we could just have this be 

```php
public ...(AuthService $authService) {
```